Library/Repository/generic/BaseRepository.cs

```diff
@@ -0,0 +1,60 @@
+﻿using DataAccessLayer.EntityDB;
+using ExceptionHandling.Exceptions;
+
+namespace DataAccessLayer.Repository
+{
+    public class BaseRepository<T> : IRepository<T> where T : class
```

Comment:
```suggestion
    public class BaseRepository<T> : IRepository<T> where T : BaseEntity
```

where BaseEntity it is a class:

public abstract class BaseEntity
{
   public int Id {get; set:}
}